engine/src/Branch.java

```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
     public Commit getStartCommit(){
         return startCommit;
     }
+
+    public void open() {
```

Comment:
Our `BranchManager` knows to `commit` the current branch and I think that it also have to know how to `loadFromCommit(String commitSHA1)`, and also all the objects that commit themselves need to know how deploy themselves and create coresponding objects from that deploy.

Lets talk about that

Reply:
the checkout action is between branches as the user experience it (behind the scene of course its switch between commits)
I think we should stay this method as branch method